File: backend/app/models/project.rb
class Project < ApplicationRecord
  belongs_to :client

  enum :status, { active: 0, maintenance: 1, canceled: 2 }

  validates :name, presence: { message: "El nombre del proyecto es requerido" }
  validates :status, presence: true

  def self.ransackable_attributes(_auth_object = nil)
    %w[id name status production_url start_date client_id created_at updated_at]
  end

  def self.ransackable_associations(_auth_object = nil)
    %w[client]
  end
end

File: frontend/src/types/project.ts
export interface Project {
  id: number;
  client_id: number;
  client_name: string;
  name: string;
  production_url: string;
  start_date: string;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface ProjectFormData {
  client_id: number | string;
  name: string;
  production_url: string;
  start_date: string;
  status: string;
}

export const PROJECT_STATUSES = [
  { value: "active", label: "Activo" },
  { value: "maintenance", label: "Mantenimiento" },
  { value: "canceled", label: "Cancelado" },
] as const;
